src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Retargeting/SdkInstallationService.cs

```diff
@@ -0,0 +1,125 @@
+    {
+        _fileSystem = fileSystem;
+        _registry = registry;
+        _environment = environment;
+    }
+
+    /// <inheritdoc/>
+    public async Task<bool> IsSdkInstalledAsync(string sdkVersion)
+    {
+        try
+        {
+            string? dotnetPath = GetDotNetPath();
+            if (dotnetPath is null)
+            {
+                return false;
+            }
+
+            // Run dotnet --list-sdks to get the list of installed SDKs
+            var startInfo = new System.Diagnostics.ProcessStartInfo
+            {
+                FileName = dotnetPath,
+                Arguments = "--list-sdks",
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                UseShellExecute = false,
+                CreateNoWindow = true
+            };
+
+            using var process = System.Diagnostics.Process.Start(startInfo);
+            if (process is null)
+            {
+                return false;
+            }
+
+            string output = await process.StandardOutput.ReadToEndAsync();
+            
+            await process.WaitForExitAsync();
+
+            if (process.ExitCode != 0)
+            {
+                return false;
+            }
+
+            // Parse the output to check if the SDK version is installed
+            // Output format: "10.0.100 [C:\Program Files\dotnet\sdk]"
+            using var reader = new StringReader(output);
+            string? line;
+            while ((line = await reader.ReadLineAsync()) is not null)
+            {
+                // Extract the version number (first part before the space)
+                int spaceIndex = line.IndexOf(' ');
+                if (spaceIndex > 0)
+                {
+                    string installedVersion = line.Substring(0, spaceIndex);
+                    if (string.Equals(installedVersion, sdkVersion, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+        catch
+        {
+            // If we fail to check, assume the SDK is not installed
+            return false;
+        }
+    }
+
+    /// <inheritdoc/>
+    public string? GetDotNetPath()
+    {
+        // First check the registry
+        string archSubKey = _environment.Is64BitOperatingSystem ? "x64" : "x86";
+        string registryKey = $@"SOFTWARE\dotnet\Setup\InstalledVersions\{archSubKey}";
+
+        string? installLocation = _registry.GetValue(
+            Win32.RegistryHive.LocalMachine,
+            Win32.RegistryView.Registry32,
+            registryKey,
+            "InstallLocation");
+
+        if (!string.IsNullOrEmpty(installLocation))
+        {
+            string dotnetExePath = Path.Combine(installLocation, "dotnet.exe");
+            if (_fileSystem.FileExists(dotnetExePath))
+            {
+                return dotnetExePath;
+            }
+        }
```

Comment:
Similar in spirit to the code in `NetCoreRuntimeVersionsRegistryReader`. Maybe we could consolidate the logic into a single MEF component.